<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Principal components analysis on the genes — genespca • pcaExplorer</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Principal components analysis on the genes — genespca" />
<meta property="og:description" content="Computes and plots the principal components of the genes, eventually displaying
the samples as in a typical biplot visualization." />






<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">pcaExplorer</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.17.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/pcaExplorer.html">User guide</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/upandrunning.html">Up and running with `r Biocpkg("pcaExplorer")`
</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/federicomarini/pcaExplorer">
    <span class="fas fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Principal components analysis on the genes</h1>
    <small class="dont-index">Source: <a href='https://github.com/federicomarini/pcaExplorer/blob/master/R/genespca.R'><code>R/genespca.R</code></a></small>
    <div class="hidden name"><code>genespca.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Computes and plots the principal components of the genes, eventually displaying
the samples as in a typical biplot visualization.</p>
    </div>

    <pre class="usage"><span class='fu'>genespca</span><span class='op'>(</span>
  <span class='va'>x</span>,
  <span class='va'>ntop</span>,
  choices <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>, <span class='fl'>2</span><span class='op'>)</span>,
  arrowColors <span class='op'>=</span> <span class='st'>"steelblue"</span>,
  groupNames <span class='op'>=</span> <span class='st'>"group"</span>,
  biplot <span class='op'>=</span> <span class='cn'>TRUE</span>,
  scale <span class='op'>=</span> <span class='fl'>1</span>,
  pc.biplot <span class='op'>=</span> <span class='cn'>TRUE</span>,
  obs.scale <span class='op'>=</span> <span class='fl'>1</span> <span class='op'>-</span> <span class='va'>scale</span>,
  var.scale <span class='op'>=</span> <span class='va'>scale</span>,
  groups <span class='op'>=</span> <span class='cn'>NULL</span>,
  ellipse <span class='op'>=</span> <span class='cn'>FALSE</span>,
  ellipse.prob <span class='op'>=</span> <span class='fl'>0.68</span>,
  labels <span class='op'>=</span> <span class='cn'>NULL</span>,
  labels.size <span class='op'>=</span> <span class='fl'>3</span>,
  alpha <span class='op'>=</span> <span class='fl'>1</span>,
  var.axes <span class='op'>=</span> <span class='cn'>TRUE</span>,
  circle <span class='op'>=</span> <span class='cn'>FALSE</span>,
  circle.prob <span class='op'>=</span> <span class='fl'>0.69</span>,
  varname.size <span class='op'>=</span> <span class='fl'>4</span>,
  varname.adjust <span class='op'>=</span> <span class='fl'>1.5</span>,
  varname.abbrev <span class='op'>=</span> <span class='cn'>FALSE</span>,
  returnData <span class='op'>=</span> <span class='cn'>FALSE</span>,
  coordEqual <span class='op'>=</span> <span class='cn'>FALSE</span>,
  scaleArrow <span class='op'>=</span> <span class='fl'>1</span>,
  useRownamesAsLabels <span class='op'>=</span> <span class='cn'>TRUE</span>,
  point_size <span class='op'>=</span> <span class='fl'>2</span>,
  annotation <span class='op'>=</span> <span class='cn'>NULL</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>x</th>
      <td><p>A <code><a href='https://rdrr.io/pkg/DESeq2/man/DESeqTransform.html'>DESeqTransform</a></code> object, with data in <code>assay(x)</code>,
produced for example by either <code><a href='https://rdrr.io/pkg/DESeq2/man/rlog.html'>rlog</a></code> or
<code><a href='https://rdrr.io/pkg/DESeq2/man/varianceStabilizingTransformation.html'>varianceStabilizingTransformation</a></code></p></td>
    </tr>
    <tr>
      <th>ntop</th>
      <td><p>Number of top genes to use for principal components,
selected by highest row variance</p></td>
    </tr>
    <tr>
      <th>choices</th>
      <td><p>Vector of two numeric values, to select on which principal components to plot</p></td>
    </tr>
    <tr>
      <th>arrowColors</th>
      <td><p>Vector of character, either as long as the number of the samples, or one single value</p></td>
    </tr>
    <tr>
      <th>groupNames</th>
      <td><p>Factor containing the groupings for the input data. Is efficiently chosen
as the (interaction of more) factors in the colData for the object provided</p></td>
    </tr>
    <tr>
      <th>biplot</th>
      <td><p>Logical, whether to additionally draw the samples labels as in a biplot representation</p></td>
    </tr>
    <tr>
      <th>scale</th>
      <td><p>Covariance biplot (scale = 1), form biplot (scale = 0). When scale = 1,
the inner product between the variables approximates the covariance and the
distance between the points approximates the Mahalanobis distance.</p></td>
    </tr>
    <tr>
      <th>pc.biplot</th>
      <td><p>Logical, for compatibility with biplot.princomp()</p></td>
    </tr>
    <tr>
      <th>obs.scale</th>
      <td><p>Scale factor to apply to observations</p></td>
    </tr>
    <tr>
      <th>var.scale</th>
      <td><p>Scale factor to apply to variables</p></td>
    </tr>
    <tr>
      <th>groups</th>
      <td><p>Optional factor variable indicating the groups that the observations
belong to. If provided the points will be colored according to groups</p></td>
    </tr>
    <tr>
      <th>ellipse</th>
      <td><p>Logical, draw a normal data ellipse for each group</p></td>
    </tr>
    <tr>
      <th>ellipse.prob</th>
      <td><p>Size of the ellipse in Normal probability</p></td>
    </tr>
    <tr>
      <th>labels</th>
      <td><p>optional Vector of labels for the observations</p></td>
    </tr>
    <tr>
      <th>labels.size</th>
      <td><p>Size of the text used for the labels</p></td>
    </tr>
    <tr>
      <th>alpha</th>
      <td><p>Alpha transparency value for the points (0 = transparent, 1 = opaque)</p></td>
    </tr>
    <tr>
      <th>var.axes</th>
      <td><p>Logical, draw arrows for the variables?</p></td>
    </tr>
    <tr>
      <th>circle</th>
      <td><p>Logical, draw a correlation circle? (only applies when prcomp
was called with scale = TRUE and when var.scale = 1)</p></td>
    </tr>
    <tr>
      <th>circle.prob</th>
      <td><p>Size of the correlation circle in Normal probability</p></td>
    </tr>
    <tr>
      <th>varname.size</th>
      <td><p>Size of the text for variable names</p></td>
    </tr>
    <tr>
      <th>varname.adjust</th>
      <td><p>Adjustment factor the placement of the variable names,
&gt;= 1 means farther from the arrow</p></td>
    </tr>
    <tr>
      <th>varname.abbrev</th>
      <td><p>Logical, whether or not to abbreviate the variable names</p></td>
    </tr>
    <tr>
      <th>returnData</th>
      <td><p>Logical, if TRUE returns a data.frame for further use, containing the
selected principal components for custom plotting</p></td>
    </tr>
    <tr>
      <th>coordEqual</th>
      <td><p>Logical, default FALSE, for allowing brushing. If TRUE, plot using
equal scale cartesian coordinates</p></td>
    </tr>
    <tr>
      <th>scaleArrow</th>
      <td><p>Multiplicative factor, usually &gt;=1, only for visualization purposes,
to allow for distinguishing where the variables are plotted</p></td>
    </tr>
    <tr>
      <th>useRownamesAsLabels</th>
      <td><p>Logical, if TRUE uses the row names as labels for plotting</p></td>
    </tr>
    <tr>
      <th>point_size</th>
      <td><p>Size of the points to be plotted for the observations (genes)</p></td>
    </tr>
    <tr>
      <th>annotation</th>
      <td><p>A <code>data.frame</code> object, with row.names as gene identifiers (e.g. ENSEMBL ids)
and a column, <code>gene_name</code>, containing e.g. HGNC-based gene symbols</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>An object created by <code>ggplot</code>, which can be assigned and further customized.</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>The implementation of this function is based on the beautiful <code>ggbiplot</code>
package developed by Vince Vu, available at https://github.com/vqv/ggbiplot.
The adaptation and additional parameters are tailored to display typical genomics data
such as the transformed counts of RNA-seq experiments</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://github.com/mikelove/DESeq2'>DESeq2</a></span><span class='op'>)</span>
<span class='va'>dds</span> <span class='op'>&lt;-</span> <span class='fu'><a href='makeExampleDESeqDataSet_multifac.html'>makeExampleDESeqDataSet_multifac</a></span><span class='op'>(</span>betaSD_condition <span class='op'>=</span> <span class='fl'>3</span>, betaSD_tissue <span class='op'>=</span> <span class='fl'>1</span><span class='op'>)</span>
<span class='va'>rlt</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/DESeq2/man/rlog.html'>rlogTransformation</a></span><span class='op'>(</span><span class='va'>dds</span><span class='op'>)</span>
<span class='va'>groups</span> <span class='op'>&lt;-</span> <span class='fu'>colData</span><span class='op'>(</span><span class='va'>dds</span><span class='op'>)</span><span class='op'>$</span><span class='va'>condition</span>
<span class='va'>groups</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/factor.html'>factor</a></span><span class='op'>(</span><span class='va'>groups</span>, levels <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/unique.html'>unique</a></span><span class='op'>(</span><span class='va'>groups</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>cols</span> <span class='op'>&lt;-</span> <span class='fu'>scales</span><span class='fu'>::</span><span class='fu'><a href='https://scales.r-lib.org//reference/hue_pal.html'>hue_pal</a></span><span class='op'>(</span><span class='op'>)</span><span class='op'>(</span><span class='fl'>2</span><span class='op'>)</span><span class='op'>[</span><span class='va'>groups</span><span class='op'>]</span>
<span class='fu'>genespca</span><span class='op'>(</span><span class='va'>rlt</span>, ntop<span class='op'>=</span><span class='fl'>100</span>, arrowColors <span class='op'>=</span> <span class='va'>cols</span>, groupNames <span class='op'>=</span> <span class='va'>groups</span><span class='op'>)</span>
</div><div class='img'><img src='genespca-1.png' alt='' width='700' height='433' /></div><div class='input'>
<span class='va'>groups_multi</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/interaction.html'>interaction</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/as.data.frame.html'>as.data.frame</a></span><span class='op'>(</span><span class='fu'>colData</span><span class='op'>(</span><span class='va'>rlt</span><span class='op'>)</span><span class='op'>[</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"condition"</span>, <span class='st'>"tissue"</span><span class='op'>)</span><span class='op'>]</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>groups_multi</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/factor.html'>factor</a></span><span class='op'>(</span><span class='va'>groups_multi</span>, levels <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/unique.html'>unique</a></span><span class='op'>(</span><span class='va'>groups_multi</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>cols_multi</span> <span class='op'>&lt;-</span> <span class='fu'>scales</span><span class='fu'>::</span><span class='fu'><a href='https://scales.r-lib.org//reference/hue_pal.html'>hue_pal</a></span><span class='op'>(</span><span class='op'>)</span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/length.html'>length</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/levels.html'>levels</a></span><span class='op'>(</span><span class='va'>groups_multi</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span><span class='op'>[</span><span class='fu'><a href='https://rdrr.io/r/base/factor.html'>factor</a></span><span class='op'>(</span><span class='va'>groups_multi</span><span class='op'>)</span><span class='op'>]</span>
<span class='fu'>genespca</span><span class='op'>(</span><span class='va'>rlt</span>, ntop <span class='op'>=</span> <span class='fl'>100</span>, arrowColors <span class='op'>=</span> <span class='va'>cols_multi</span>, groupNames <span class='op'>=</span> <span class='va'>groups_multi</span><span class='op'>)</span>
</div><div class='img'><img src='genespca-2.png' alt='' width='700' height='433' /></div><div class='input'>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by <a href='https://federicomarini.github.io'>Federico Marini</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.9000.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


