<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>pcaExplorer User Guide • pcaExplorer</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="pcaExplorer User Guide">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">pcaExplorer</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.9.6</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/pcaExplorer.html">User guide</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/upandrunning.html">Up and running with `r Biocpkg("pcaExplorer")`
</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/federicomarini/pcaExplorer">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><link href="pcaExplorer_files/font-awesome-5.1.0/css/all.css" rel="stylesheet">
<link href="pcaExplorer_files/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet">
<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>pcaExplorer User Guide</h1>
                        <h4 class="author">Federico Marini</h4>
            <address class="author_afil">
      Institute of Medical Biostatistics, Epidemiology and Informatics (IMBEI), Mainz<br><a class="author_email" href="mailto:#"></a><a href="mailto:marinif@uni-mainz.de" class="email">marinif@uni-mainz.de</a>
      </address>
                              <h4 class="author">Harald Binder</h4>
            <address class="author_afil">
      Institute of Medical Biostatistics, Epidemiology and Informatics (IMBEI), Mainz<br><h4 class="date">8 April 2019</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/federicomarini/pcaExplorer/blob/master/vignettes/pcaExplorer.Rmd"><code>vignettes/pcaExplorer.Rmd</code></a></small>
      <div class="hidden name"><code>pcaExplorer.Rmd</code></div>

    </address>
</div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <em><a href="https://bioconductor.org/packages/3.9/pcaExplorer">pcaExplorer</a></em> provides functionality for interactive visualization of RNA-seq datasets based on Principal Components Analysis. Such methods allow for quick information extraction and effective data exploration. A Shiny application encapsulates the whole analysis, and is developed to become a practical companion for any RNA-seq dataset. This app supports reproducible research with state saving and automated report generation.
    </div>
    
<p><br><strong>Package</strong>: <em>pcaExplorer</em><br><strong>Authors</strong>: Federico Marini [aut, cre] (<a href="https://orcid.org/0000-0003-3252-7758" class="uri">https://orcid.org/0000-0003-3252-7758</a>)<br><strong>Version</strong>: 2.9.5<br><strong>Compiled date</strong>: 2019-04-08<br><strong>Last edited</strong>: 2019-02-26<br><strong>License</strong>: MIT + file LICENSE<br></p>
<script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  document.querySelector("h1").className = "title";
});
</script><script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  var links = document.links;  
  for (var i = 0, linksLength = links.length; i < linksLength; i++)
    if (links[i].hostname != window.location.hostname)
      links[i].target = '_blank';
});
</script><div id="getting-started" class="section level1">
<h1 class="hasAnchor">
<a href="#getting-started" class="anchor"></a>Getting started</h1>
<p><em><a href="https://bioconductor.org/packages/3.9/pcaExplorer">pcaExplorer</a></em> is an R package distributed as part of the <a href="http://bioconductor.org">Bioconductor</a> project. To install the package, start R and enter:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="cf">if</span> (<span class="op">!</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/ns-load">requireNamespace</a></span>(<span class="st">"BiocManager"</span>, <span class="dt">quietly=</span><span class="ot">TRUE</span>))</a>
<a class="sourceLine" id="cb1-2" data-line-number="2">    <span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/install.packages">install.packages</a></span>(<span class="st">"BiocManager"</span>)</a>
<a class="sourceLine" id="cb1-3" data-line-number="3">BiocManager<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/BiocManager/topics/install">install</a></span>(<span class="st">"pcaExplorer"</span>)</a></code></pre></div>
<p>To install <em><a href="https://bioconductor.org/packages/3.9/pcaExplorer">pcaExplorer</a></em> with all its dependencies (i.e. also the ones listed in the <code>Suggests:</code> field of the <code>DESCRIPTION</code> file, which include the dataset from the <em><a href="https://bioconductor.org/packages/3.9/airway">airway</a></em> package used as a demo), use this command instead:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1">BiocManager<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/BiocManager/topics/install">install</a></span>(<span class="st">"pcaExplorer"</span>, <span class="dt">dependencies =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<p>If you prefer, you can install and use the development version, which can be retrieved via Github (<a href="https://github.com/federicomarini/pcaExplorer" class="uri">https://github.com/federicomarini/pcaExplorer</a>). To do so, use:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(<span class="st">"devtools"</span>)</a>
<a class="sourceLine" id="cb3-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/devtools/topics/reexports">install_github</a></span>(<span class="st">"federicomarini/pcaExplorer"</span>)</a></code></pre></div>
<p>Once <em>pcaExplorer</em> is installed, it can be loaded by the following command.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(<span class="st">"pcaExplorer"</span>)</a></code></pre></div>
</div>
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p><em><a href="https://bioconductor.org/packages/3.9/pcaExplorer">pcaExplorer</a></em> is a Bioconductor package containing a Shiny application for analyzing expression data in different conditions and experimental factors.</p>
<p>It is a general-purpose interactive companion tool for RNA-seq analysis, which guides the user in exploring the Principal Components of the data under inspection.</p>
<p><em><a href="https://bioconductor.org/packages/3.9/pcaExplorer">pcaExplorer</a></em> provides tools and functionality to detect outlier samples, genes that show particular patterns, and additionally provides a functional interpretation of the principal components for further quality assessment and hypothesis generation on the input data.</p>
<p>Moreover, a novel visualization approach is presented to simultaneously assess the effect of more than one experimental factor on the expression levels.</p>
<p>Thanks to its interactive/reactive design, it is designed to become a practical companion to any RNA-seq dataset analysis, making exploratory data analysis accessible also to the bench biologist, while providing additional insight also for the experienced data analyst.</p>
<p>Starting from development version 1.1.3, <em><a href="https://bioconductor.org/packages/3.9/pcaExplorer">pcaExplorer</a></em> supports reproducible research with state saving and automated report generation. Each generated plot and table can be exported by simple mouse clicks on the dedicated buttons.</p>
<div id="citation-info" class="section level2">
<h2 class="hasAnchor">
<a href="#citation-info" class="anchor"></a>Citation info</h2>
<p>If you use <em><a href="https://bioconductor.org/packages/3.9/pcaExplorer">pcaExplorer</a></em> for your analysis, please cite it as here below:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/citation">citation</a></span>(<span class="st">"pcaExplorer"</span>)</a></code></pre></div>
<pre><code>
To cite the package 'pcaExplorer' in publications use:

  Federico Marini, Harald Binder (2016). Development of
  Applications for Interactive and Reproducible Research: a Case
  Study URL http://bioconductor.org/packages/pcaExplorer/,
  https://genomicscomputbiol.org/ojs3/GCB/article/view/21 doi:
  10.18547/gcb.2017.vol3.iss1.e39, 10.18129/B9.bioc.pcaExplorer

A BibTeX entry for LaTeX users is

  @Article{,
    title = {Development of Applications for Interactive and Reproducible Research: a Case Study},
    journal = {Genomics and Computational Biology},
    author = {Federico Marini and Harald Binder},
    volume = {3},
    number = {1},
    year = {2016},
    issn = {2365-7154},
    pages = {e39},
    doi = {10.18547/gcb.2017.vol3.iss1.e39},
    url = {http://bioconductor.org/packages/pcaExplorer/},
  }</code></pre>
</div>
</div>
<div id="launching-the-application" class="section level1">
<h1 class="hasAnchor">
<a href="#launching-the-application" class="anchor"></a>Launching the application</h1>
<p>After loading the package, the <em><a href="https://bioconductor.org/packages/3.9/pcaExplorer">pcaExplorer</a></em> app can be launched in different modes:</p>
<ul>
<li><p><code><a href="../reference/pcaExplorer.html">pcaExplorer(dds = dds, dst = dst)</a></code>, where <code>dds</code> is a <code>DESeqDataSet</code> object and <code>dst</code> is a <code>DESeqTransform</code> object, which were created during an existing session for the analysis of an RNA-seq dataset with the <em><a href="https://bioconductor.org/packages/3.9/DESeq2">DESeq2</a></em> package.</p></li>
<li><p><code><a href="../reference/pcaExplorer.html">pcaExplorer(dds = dds)</a></code>, where <code>dds</code> is a <code>DESeqDataSet</code> object. The <code>dst</code> object can be automatically computed upon launch, choosing between rlog transformation, variance stabilizing transformations, or shifted logarithm transformation (with pseudocount = 1).</p></li>
<li><p><code><a href="../reference/pcaExplorer.html">pcaExplorer(countmatrix = countmatrix, coldata = coldata)</a></code>, where <code>countmatrix</code> is a count matrix, generated after assigning reads to features such as genes via tools such as <code>HTSeq-count</code> or <code>featureCounts</code>, and <code>coldata</code> is a data frame containing the experimental covariates of the experiments, such as condition, tissue, cell line, run batch and so on. If the data is provided in this way, the user can click on the “Generate the dds and dst objects” button to complete the setup and enable the subsequent steps in the other panels.</p></li>
<li><p><code><a href="../reference/pcaExplorer.html">pcaExplorer()</a></code>, and then subsequently uploading the count matrix and the covariates data frame through the user interface. These files need to be formatted as tab, semicolon, or comma separated text files, all of which are common formats for storing such count values.</p></li>
</ul>
<p>Additional parameters and objects that can be provided to the main <em><a href="https://bioconductor.org/packages/3.9/pcaExplorer">pcaExplorer</a></em> function are:</p>
<ul>
<li><p><code>pca2go</code>, which is an object created by the <code>pca2go</code> function, which scans the genes with high loadings in each principal component and each direction, and looks for functions (such as GO Biological Processes) that are enriched above the background. The offline <code>pca2go</code> function is based on the routines and algorithms of the <em><a href="https://bioconductor.org/packages/3.9/topGO">topGO</a></em> package, but as an alternative, this object can be computed live during the execution of the app with <code>limmaquickpca2go</code> (which relies on the <code>goana</code> function provided by the <em><a href="https://bioconductor.org/packages/3.9/limma">limma</a></em> package). Although this likely provides more general (and probably less informative) functions, it is a good compromise for quickly obtaining a further data interpretation.</p></li>
<li><p><code>annotation</code>, a data frame object, with <code>row.names</code> as gene identifiers (e.g. ENSEMBL ids) identical to the row names of the count matrix or <code>dds</code> object, and an extra column <code>gene_name</code>, containing e.g. HGNC-based gene symbols. This can be used for making information extraction easier, as ENSEMBL ids (a usual choice when assigning reads to features) do not provide an immediate readout for which gene they refer to. This can be either passed as a parameter when launching the app, or also uploaded as a text file (either tab, comma, or semicolon-separated). The package provides two functions, <code>get_annotation</code> and <code>get_annotation_orgdb</code>, as a convenient wrapper to obtain the updated annotation information, respectively from <code>biomaRt</code> or via the <code>org.XX.eg.db</code> packages.</p></li>
</ul>
<div id="how-to-provide-your-input-data-in-pcaexplorer" class="section level2">
<h2 class="hasAnchor">
<a href="#how-to-provide-your-input-data-in-pcaexplorer" class="anchor"></a>How to provide your input data in <em><a href="https://bioconductor.org/packages/3.9/pcaExplorer">pcaExplorer</a></em>
</h2>
<p><em><a href="https://bioconductor.org/packages/3.9/pcaExplorer">pcaExplorer</a></em> supports a number of file formats when uploading the data via the file input widgets. Starting from version 2.9.5, we added functionality to select the separator character for each of the uploadable files. An information box is also shown by clicking on the question mark icon in the Data upload panel, with detailed information (text, as well as screenshots of valid input files) on the format specification.</p>
<p>In general, <em><a href="https://bioconductor.org/packages/3.9/pcaExplorer">pcaExplorer</a></em> requires by default tab separated files:</p>
<ul>
<li>the <code>countmatrix</code>: contains the expression matrix, with one gene per row and one sample per column; the first column should contain the gene identifiers, and the header (first row) specifies the sample names.</li>
<li>the <code>coldata</code>: one sample per row, and one experimental covariate per column. Row names should be specified in the first column, and have to match the column names of the <code>countmatrix</code>. Column names will contain the specific experimental covariates.</li>
<li>the <code>annotation</code> (optional): one gene per row, and one identifier type per column. Gene identifiers in the first column are identical to the row names of the <code>countmatrix</code> or <code>dds</code> objects. At least an extra column <code>gene_name</code>, containing e.g. HGNC-based gene symbols, needs to be provided.</li>
</ul>
</div>
<div id="up-and-running-with-pcaexplorer" class="section level2">
<h2 class="hasAnchor">
<a href="#up-and-running-with-pcaexplorer" class="anchor"></a>Up and running with <em><a href="https://bioconductor.org/packages/3.9/pcaExplorer">pcaExplorer</a></em>
</h2>
<p>We recommend users to switch to the <a href="https://bioconductor.org/packages/3.9/pcaExplorer/vignettes/upandrunning.html">dedicated vignette</a>, entitled “Up and running with <em><a href="https://bioconductor.org/packages/3.9/pcaExplorer">pcaExplorer</a></em>”.</p>
<p>This document describes a use case for <em><a href="https://bioconductor.org/packages/3.9/pcaExplorer">pcaExplorer</a></em>, based on the dataset in the <em><a href="https://bioconductor.org/packages/3.9/airway">airway</a></em> package.</p>
</div>
</div>
<div id="the-controls-sidebar" class="section level1">
<h1 class="hasAnchor">
<a href="#the-controls-sidebar" class="anchor"></a>The controls sidebar</h1>
<p>Most of the input controls are located in the sidebar, some are as well in the individual tabs of the app. By changing one or more of the input parameters, the user can get a fine control on what is displayed.</p>
<div id="app-settings" class="section level2">
<h2 class="hasAnchor">
<a href="#app-settings" class="anchor"></a>App settings</h2>
<p>Here are the parameters that set input values for most of the tabs. By hovering over with the mouse, the user can receive additional information on how to set the parameter, with tooltips powered by the <em><a href="https://CRAN.R-project.org/package=shinyBS">shinyBS</a></em> package.</p>
<ul>
<li>
<strong>x-axis PC</strong> - Select the principal component to display on the x axis</li>
<li>
<strong>y-axis PC</strong> - Select the principal component to display on the y axis</li>
<li>
<strong>Group/color by</strong> - Select the group of samples to stratify the analysis. Can also assume multiple values.</li>
<li>
<strong>Nr of (most variant) genes</strong> - Number of genes to select for computing the principal components. The top n genes are selected ranked by their variance inter-samples</li>
<li>
<strong>Alpha</strong> - Color transparency for the plots. Can assume values from 0 (transparent) to 1 (opaque)</li>
<li>
<strong>Labels size</strong> - Size of the labels for the samples in the principal components plots. This parameter also controls the size of the gene labels, which are displayed in the Genes View once the user has brushed an area in the main plot.</li>
<li>
<strong>Points size</strong> - Size of the points to be plotted in the principal components plots</li>
<li>
<strong>Variable name size</strong> - Size of the labels for the genes PCA - correspond to the samples names</li>
<li>
<strong>Scaling factor</strong> - Scale value for resizing the arrow corresponding to the variables in the PCA for the genes. It should be used for mere visualization purposes</li>
<li>
<strong>Color palette</strong> - Select the color palette to be used in the principal components plots. The number of colors is selected automatically according to the number of samples and to the levels of the factors of interest and their interactions</li>
<li>
<strong>Plot style for gene counts</strong> - Plot either boxplots or violin plots, with jittered points superimposed</li>
</ul>
</div>
<div id="plot-export-settings" class="section level2">
<h2 class="hasAnchor">
<a href="#plot-export-settings" class="anchor"></a>Plot export settings</h2>
<p><strong>Width</strong> and <strong>height</strong> for the figures to export are input here in cm.</p>
<p>Additional controls available in the single tabs are also assisted by tooltips that show on hovering the mouse. Normally they are tightly related to the plot/output they are placed nearby.</p>
</div>
</div>
<div id="the-task-menu" class="section level1">
<h1 class="hasAnchor">
<a href="#the-task-menu" class="anchor"></a>The task menu</h1>
<p>The task menu, accessible by clicking on the cog icon in the upper right part of the application, provides two functionalities:</p>
<ul>
<li>
<code>Exit pcaExplorer &amp; save</code> will close the application and store the content of the <code>input</code> and <code>values</code> reactive objects in two list objects made available in the global environment, called <code>pcaExplorer_inputs_YYYYMMDD_HHMMSS</code> and <code>pcaExplorer_values_YYYYMMDD_HHMMSS</code>
</li>
<li>
<code>Save State as .RData</code> will similarly store <code>LiveInputs</code> and <code>r_data</code> in a binary file named <code>pcaExplorerState_YYYYMMDD_HHMMSS.Rdata</code>, without closing the application</li>
</ul>
</div>
<div id="the-app-panels" class="section level1">
<h1 class="hasAnchor">
<a href="#the-app-panels" class="anchor"></a>The app panels</h1>
<p>The <em><a href="https://bioconductor.org/packages/3.9/pcaExplorer">pcaExplorer</a></em> app is structured in different panels, each focused on a different aspect of the data exploration.</p>
<p>Most of the panels work extensively with click-based and brush-based interactions, to gain additional depth in the explorations, for example by zooming, subsetting, selecting. This is possible thanks to the recent developments in the <em><a href="https://CRAN.R-project.org/package=shiny">shiny</a></em> package/framework.</p>
<p>The available panels are described in the following subsections.</p>
<div id="data-upload" class="section level2">
<h2 class="hasAnchor">
<a href="#data-upload" class="anchor"></a>Data Upload</h2>
<p>These <strong>file input</strong> controls are available when no <code>dds</code> or <code>countmatrix</code> + <code>coldata</code> are provided. Additionally, it is possible to upload the <code>annotation</code> data frame. If the objects are already passed as parameters, or after they have been successfully uploaded, a brief overview/summary for them can be displayed, by clicking on each respective action button.</p>
<p align="center">
<img src="newsnap_01_upload.png" style="width:1000px;"></p>
<p>This panel is where you can perform the preprocessing steps on the data you uploaded/provided:</p>
<ul>
<li>compose the <code>dds</code> object (if you provided <code>countmatrix</code> and <code>coldata</code>)</li>
<li>normalize the expression values (using the robust method proposed by Anders and Huber in the original DESeq manuscript)</li>
<li>compute the variance stabilizing transformed expression values (stored in the <code>dst</code> object).</li>
</ul>
<p>As a note regarding the normalization procedure: the normalization method (implemented in <code>estimateSizeFactors</code>) relies on the hypothesis that most of the genes are not differentially expressed across experimental groups, and this holds true for the majority of scenarios. The <code>DESeqDataSet</code> object, which pcaExplorer takes as main data container, can still accommodate sample (and gene) specific normalization factors. Should this assumption be violated, users can pre-compute these factors and store them in the input <code>dds</code> object.</p>
</div>
<div id="instructions" class="section level2">
<h2 class="hasAnchor">
<a href="#instructions" class="anchor"></a>Instructions</h2>
<p>This is where you might be reading a version of the “Up and running with <em><a href="https://bioconductor.org/packages/3.9/pcaExplorer">pcaExplorer</a></em>” vignette. Additionally, you can easily reach the fully rendered vignettes, either installed locally, or directly from the Bioconductor package page.</p>
<p align="center">
<img src="newsnap_02_instructions.png" style="width:1000px;"></p>
</div>
<div id="counts-table" class="section level2">
<h2 class="hasAnchor">
<a href="#counts-table" class="anchor"></a>Counts Table</h2>
<p>Interactive tables for the raw, normalized or transformed (rlog, variance stabilized, or shifted log) counts are shown in this tab. The user can also generate a sample-to-sample correlation scatter plot with the selected data.</p>
<p align="center">
<img src="newsnap_03_countstable.png" style="width:1000px;"></p>
</div>
<div id="data-overview" class="section level2">
<h2 class="hasAnchor">
<a href="#data-overview" class="anchor"></a>Data Overview</h2>
<p>This panel displays information on the objects in use, either passed as parameters or generated from the count matrix provided. Displayed information comprise the design metadata, a sample to sample distance heatmap, the number of million of reads per sample and some basic summary for the counts.</p>
<p align="center">
<img src="newsnap_04_overview.png" style="width:1000px;"></p>
</div>
<div id="samples-view" class="section level2">
<h2 class="hasAnchor">
<a href="#samples-view" class="anchor"></a>Samples View</h2>
<p>This panel displays the PCA projections of sample expression profiles onto any pair of components, a scree plot, a zoomed PCA plot, a plot of the genes with top and bottom loadings. Additionally, this section presents a PCA plot where it is possible to remove samples deemed to be outliers in the analysis, which is very useful to check the effect of excluding them. If needed, an interactive 3D visualization of the principal components is also available.</p>
<p align="center">
<img src="newsnap_05_samples.png" style="width:1000px;"></p>
</div>
<div id="genes-view" class="section level2">
<h2 class="hasAnchor">
<a href="#genes-view" class="anchor"></a>Genes View</h2>
<p>This panel displays the PCA projections of genes abundances onto any pair of components, with samples as biplot variables, to identify interesting groups of genes. Zooming is also possible, and clicking on single genes, a boxplot is returned, grouped by the factors of interest. A static and an interactive heatmap are provided, including the subset of selected genes, also displayed as (standardized) expression profiles across the samples. These are also reported in <code>datatable</code> objects, accessible in the bottom part of the tab.</p>
<p align="center">
<img src="newsnap_06_genes.png" style="width:1000px;"></p>
</div>
<div id="genefinder" class="section level2">
<h2 class="hasAnchor">
<a href="#genefinder" class="anchor"></a>GeneFinder</h2>
<p>The user can search and display the expression values of a gene of interest, either by ID or gene name, as provided in the <code>annotation</code>. A handy panel for quick screening of shortlisted genes, again grouped by the factors of interest. The graphic can be readily exported as it is, and this can be iterated on a shortlisted set of genes. For each of them, the underlying data is displayed in an interactive table, also exportable with a click.</p>
<p align="center">
<img src="newsnap_07_finder.png" style="width:1000px;"></p>
</div>
<div id="pca2go" class="section level2">
<h2 class="hasAnchor">
<a href="#pca2go" class="anchor"></a>PCA2GO</h2>
<p>This panel shows the functional annotation of the principal components, with GO functions enriched in the genes with high loadings on the selected principal components. It allows for the live computing of the object, that can otherwise provided as a parameter when launching the app. The panel displays a PCA plot for the samples, surrounded on each side by the tables with the functions enriched in each component and direction.</p>
<p align="center">
<img src="newsnap_08_pca2go.png" style="width:1000px;"></p>
<div id="more-on-the-pca2go-parameter" class="section level3">
<h3 class="hasAnchor">
<a href="#more-on-the-pca2go-parameter" class="anchor"></a>More on the <code>pca2go</code> parameter</h3>
<p>A note on the functionality provided by the PCA2GO tab: if it is not provided in the <code>pca2go</code> parameter, the user can still compute this object while using the app, and this is done using <code><a href="https://www.rdocumentation.org/packages/limma/topics/goana">limma::goana</a></code>. The implementation of this function supports a few organism packages, which cover many use cases (human, mouse, rat, fruit fly, or chimpanzee), but some use cases are not directly covered (e.g. if you are working with plants like Arabidopsis Thaliana).</p>
<p>For example, with a dataset from Arabidopsis where the genes are encoded as TAIR (The Arabidopsis Information Resource) identifiers, one would ideally call the following commands</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1">BiocManager<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/BiocManager/topics/install">install</a></span>(<span class="st">"org.At.tair.db"</span>)</a>
<a class="sourceLine" id="cb7-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(<span class="st">"org.At.tair.db"</span>)</a>
<a class="sourceLine" id="cb7-3" data-line-number="3"><span class="co"># skipping the steps where you normally would generate your dds_at object...</span></a>
<a class="sourceLine" id="cb7-4" data-line-number="4">dds_at</a>
<a class="sourceLine" id="cb7-5" data-line-number="5">vst_at &lt;-<span class="st"> </span>DESeq2<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/DESeq2/topics/vst">vst</a></span>(dds_at)</a>
<a class="sourceLine" id="cb7-6" data-line-number="6">anno_at &lt;-<span class="st"> </span><span class="kw"><a href="../reference/get_annotation_orgdb.html">get_annotation_orgdb</a></span>(dds_at,<span class="st">"org.At.tair.db"</span>, <span class="dt">idtype =</span> <span class="st">"TAIR"</span>)</a>
<a class="sourceLine" id="cb7-7" data-line-number="7"><span class="co"># subset the background to include only the expressed genes</span></a>
<a class="sourceLine" id="cb7-8" data-line-number="8">bg_ids &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">rownames</a></span>(dds_at)[<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colSums">rowSums</a></span>(<span class="kw">counts</span>(dds_at)) <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>]</a>
<a class="sourceLine" id="cb7-9" data-line-number="9"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(topGO)</a>
<a class="sourceLine" id="cb7-10" data-line-number="10">pca2go_at &lt;-<span class="st"> </span><span class="kw"><a href="../reference/pca2go.html">pca2go</a></span>(vst_at,</a>
<a class="sourceLine" id="cb7-11" data-line-number="11">                    <span class="dt">annotation =</span> anno_at,</a>
<a class="sourceLine" id="cb7-12" data-line-number="12">                    <span class="dt">annopkg =</span> <span class="st">"org.At.tair.db"</span>,</a>
<a class="sourceLine" id="cb7-13" data-line-number="13">                    <span class="dt">ensToGeneSymbol =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb7-14" data-line-number="14">                    <span class="dt">background_genes =</span> bg_ids)</a>
<a class="sourceLine" id="cb7-15" data-line-number="15"><span class="co"># and finally, with all the objects prepared...</span></a>
<a class="sourceLine" id="cb7-16" data-line-number="16"><span class="kw"><a href="../reference/pcaExplorer.html">pcaExplorer</a></span>(<span class="dt">dds =</span> dds_at, <span class="dt">dst =</span> vst_at, <span class="dt">annotation =</span> anno_at, <span class="dt">pca2go =</span> pca2go_at)</a></code></pre></div>
</div>
</div>
<div id="multifactor-exploration" class="section level2">
<h2 class="hasAnchor">
<a href="#multifactor-exploration" class="anchor"></a>Multifactor Exploration</h2>
<p>This panel allows for the multifactor exploration of datasets with 2 or more experimental factors. The user has to select first the two factors and the levels for each. Then, it is possible to combine samples from Factor1-Level1 in the selected order by clicking on each sample name, one for each level available in the selected Factor2. In order to build the matrix, an equal number of samples for each level of Factor 1 is required, to keep the design somehow balanced. A typical case for choosing factors 1 and 2 is for example when different conditions and tissues are present.</p>
<p>Once constructed, a plot is returned that tries to represent simultaneously the effect of the two factors on the data. Each gene is represented by a dot-line-dot structure, with the color that is indicating the tissue (factor 2) where the gene is mostly expressed. Each gene has two dots, one for each condition level (factor 1), and the position of the points is dictated by the scores of the principal components calculated on the matrix object. The line connecting the dots is darker when the tissue where the gene is mostly expressed varies throughout the conditions.</p>
<p>This representation is under active development, and it is promising for identifying interesting sets or clusters of genes according to their behavior on the Principal Components subspaces. Zooming and exporting of the underlying genes is also allowed by brushing on the main plot.</p>
<p align="center">
<img src="newsnap_09_multifac.png" style="width:1000px;"></p>
</div>
<div id="report-editor" class="section level2">
<h2 class="hasAnchor">
<a href="#report-editor" class="anchor"></a>Report Editor</h2>
<p>The report editor is the backbone for generating and editing the interactive report on the basis of the uploaded data and the current state of the application. General <code>Markdown options</code> and <code>Editor options</code> are available, and the text editor, based on the <code>shinyAce</code> package, contains a comprehensive template report, that can be edited to the best convenience of the user.</p>
<p>The editor supports R code autocompletion, making it easy to add new code chunks for additional sections. A preview is available in the tab itself, and the report can be generated, saved and subsequently shared with simple mouse clicks.</p>
<p align="center">
<img src="newsnap_10_editor.png" style="width:1000px;"></p>
<p>The functionality to display the report preview is based on <code>knit2html</code>, and some elements such as <code>DataTable</code> objects might not render correctly. To render them correctly, please install the PhantomJS executable before launching the app. This can be done by using the <em><a href="https://CRAN.R-project.org/package=webshot">webshot</a></em> package and calling <code><a href="https://www.rdocumentation.org/packages/webshot/topics/install_phantomjs">webshot::install_phantomjs()</a></code> - HTML widgets will be rendered automatically as screenshots. Keep in mind that the fully rendered report (the one you can obtain with the “Generate &amp; Save” button) is not affected by this, since it uses <code><a href="https://www.rdocumentation.org/packages/rmarkdown/topics/render">rmarkdown::render()</a></code>.</p>
</div>
<div id="about" class="section level2">
<h2 class="hasAnchor">
<a href="#about" class="anchor"></a>About</h2>
<p>Contains general information on <em><a href="https://bioconductor.org/packages/3.9/pcaExplorer">pcaExplorer</a></em>, including the developer’s contact, the link to the development version in Github, as well as the output of <code>sessionInfo</code>, to use for reproducibility sake - or bug reporting. Information for citing <em><a href="https://bioconductor.org/packages/3.9/pcaExplorer">pcaExplorer</a></em> is also reported.</p>
<p align="center">
<img src="newsnap_11_about.png" style="width:1000px;"></p>
</div>
</div>
<div id="running-pcaexplorer-on-published-datasets" class="section level1">
<h1 class="hasAnchor">
<a href="#running-pcaexplorer-on-published-datasets" class="anchor"></a>Running <code>pcaExplorer</code> on published datasets</h1>
<p>We can run <em><a href="https://bioconductor.org/packages/3.9/pcaExplorer">pcaExplorer</a></em> for demonstration purpose on published datasets that are available as SummarizedExperiment in a Bioconductor experiment package.</p>
<p>We will use the <em><a href="https://bioconductor.org/packages/3.9/airway">airway</a></em> dataset, which can be installed with this command:</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1"><span class="cf">if</span> (<span class="op">!</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/ns-load">requireNamespace</a></span>(<span class="st">"BiocManager"</span>, <span class="dt">quietly=</span><span class="ot">TRUE</span>))</a>
<a class="sourceLine" id="cb8-2" data-line-number="2">    <span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/install.packages">install.packages</a></span>(<span class="st">"BiocManager"</span>)</a>
<a class="sourceLine" id="cb8-3" data-line-number="3">BiocManager<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/BiocManager/topics/install">install</a></span>(<span class="st">"airway"</span>)</a></code></pre></div>
<p>The <em><a href="https://bioconductor.org/packages/3.9/airway">airway</a></em> package provides a <code>RangedSummarizedExperiment</code> object of read counts in genes for an RNA-Seq experiment on four human airway smooth muscle cell lines treated with dexamethasone. More details such as gene models and count quantifications can be found in the <em><a href="https://bioconductor.org/packages/3.9/airway">airway</a></em> package vignette.</p>
<p>The easiest way to explore the <em><a href="https://bioconductor.org/packages/3.9/airway">airway</a></em> dataset is by clicking on the dedicated button in the <strong>Data Upload</strong> panel.</p>
<button id="btn_loaddemo" type="button" class="btn btn-default action-button shiny-bound-input" style="color: #0092AC" data-original-title title>
<i class="fa fa-play-circle"></i> Load the demo airway data
</button>
<p>Otherwise, to run <em><a href="https://bioconductor.org/packages/3.9/pcaExplorer">pcaExplorer</a></em> on this dataset from the terminal/RStudio IDE, the following commands are required. First, prepare the objects to be passed as parameters of <em><a href="https://bioconductor.org/packages/3.9/pcaExplorer">pcaExplorer</a></em>.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(airway)</a>
<a class="sourceLine" id="cb9-2" data-line-number="2"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(DESeq2)</a>
<a class="sourceLine" id="cb9-3" data-line-number="3"></a>
<a class="sourceLine" id="cb9-4" data-line-number="4"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/data">data</a></span>(airway)</a>
<a class="sourceLine" id="cb9-5" data-line-number="5"></a>
<a class="sourceLine" id="cb9-6" data-line-number="6">dds_airway &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/DESeq2/topics/DESeqDataSet">DESeqDataSet</a></span>(airway,<span class="dt">design=</span> <span class="op">~</span><span class="st"> </span>cell <span class="op">+</span><span class="st"> </span>dex)</a>
<a class="sourceLine" id="cb9-7" data-line-number="7">dds_airway</a></code></pre></div>
<pre><code>class: DESeqDataSet 
dim: 64102 8 
metadata(2): '' version
assays(1): counts
rownames(64102): ENSG00000000003 ENSG00000000005 ... LRG_98 LRG_99
rowData names(0):
colnames(8): SRR1039508 SRR1039509 ... SRR1039520 SRR1039521
colData names(9): SampleName cell ... Sample BioSample</code></pre>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1">rld_airway &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/DESeq2/topics/rlog">rlogTransformation</a></span>(dds_airway)</a>
<a class="sourceLine" id="cb11-2" data-line-number="2">rld_airway</a></code></pre></div>
<pre><code>class: DESeqTransform 
dim: 64102 8 
metadata(2): '' version
assays(1): ''
rownames(64102): ENSG00000000003 ENSG00000000005 ... LRG_98 LRG_99
rowData names(7): baseMean baseVar ... dispFit rlogIntercept
colnames(8): SRR1039508 SRR1039509 ... SRR1039520 SRR1039521
colData names(10): SampleName cell ... BioSample sizeFactor</code></pre>
<p>Then launch the app itself.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1"><span class="kw"><a href="../reference/pcaExplorer.html">pcaExplorer</a></span>(<span class="dt">dds =</span> dds_airway,</a>
<a class="sourceLine" id="cb13-2" data-line-number="2">            <span class="dt">dst =</span> rld_airway)</a></code></pre></div>
<p>The <code>annotation</code> for this dataset can be built by exploiting the <em><a href="https://bioconductor.org/packages/3.9/org.Hs.eg.db">org.Hs.eg.db</a></em> package.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(org.Hs.eg.db)</a>
<a class="sourceLine" id="cb14-2" data-line-number="2">genenames_airway &lt;-<span class="st"> </span><span class="kw">mapIds</span>(org.Hs.eg.db,<span class="dt">keys =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">rownames</a></span>(dds_airway),<span class="dt">column =</span> <span class="st">"SYMBOL"</span>,<span class="dt">keytype=</span><span class="st">"ENSEMBL"</span>)</a>
<a class="sourceLine" id="cb14-3" data-line-number="3">annotation_airway &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/data.frame">data.frame</a></span>(<span class="dt">gene_name =</span> genenames_airway,</a>
<a class="sourceLine" id="cb14-4" data-line-number="4">                                <span class="dt">row.names =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">rownames</a></span>(dds_airway),</a>
<a class="sourceLine" id="cb14-5" data-line-number="5">                                <span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb14-6" data-line-number="6"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/head">head</a></span>(annotation_airway)                                </a></code></pre></div>
<pre><code>                gene_name
ENSG00000000003    TSPAN6
ENSG00000000005      TNMD
ENSG00000000419      DPM1
ENSG00000000457     SCYL3
ENSG00000000460  C1orf112
ENSG00000000938       FGR</code></pre>
<p>or alternatively, by using the <code>get_annotation</code> or <code>get_annotation_orgdb</code> wrappers.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1">anno_df_orgdb &lt;-<span class="st"> </span><span class="kw"><a href="../reference/get_annotation_orgdb.html">get_annotation_orgdb</a></span>(<span class="dt">dds =</span> dds_airway,</a>
<a class="sourceLine" id="cb16-2" data-line-number="2">                                      <span class="dt">orgdb_species =</span> <span class="st">"org.Hs.eg.db"</span>,</a>
<a class="sourceLine" id="cb16-3" data-line-number="3">                                      <span class="dt">idtype =</span> <span class="st">"ENSEMBL"</span>)</a>
<a class="sourceLine" id="cb16-4" data-line-number="4"></a>
<a class="sourceLine" id="cb16-5" data-line-number="5">anno_df_biomart &lt;-<span class="st"> </span><span class="kw"><a href="../reference/get_annotation.html">get_annotation</a></span>(<span class="dt">dds =</span> dds_airway,</a>
<a class="sourceLine" id="cb16-6" data-line-number="6">                                  <span class="dt">biomart_dataset =</span> <span class="st">"hsapiens_gene_ensembl"</span>,</a>
<a class="sourceLine" id="cb16-7" data-line-number="7">                                  <span class="dt">idtype =</span> <span class="st">"ensembl_gene_id"</span>)</a></code></pre></div>
<pre><code>'select()' returned 1:many mapping between keys and columns</code></pre>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb18-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/head">head</a></span>(anno_df_orgdb)</a></code></pre></div>
<pre><code>                        gene_id gene_name
ENSG00000000003 ENSG00000000003    TSPAN6
ENSG00000000005 ENSG00000000005      TNMD
ENSG00000000419 ENSG00000000419      DPM1
ENSG00000000457 ENSG00000000457     SCYL3
ENSG00000000460 ENSG00000000460  C1orf112
ENSG00000000938 ENSG00000000938       FGR</code></pre>
<p>Then again, the app can be launched with:</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb20-1" data-line-number="1"><span class="kw"><a href="../reference/pcaExplorer.html">pcaExplorer</a></span>(<span class="dt">dds =</span> dds_airway,</a>
<a class="sourceLine" id="cb20-2" data-line-number="2">            <span class="dt">dst =</span> rld_airway,</a>
<a class="sourceLine" id="cb20-3" data-line-number="3">            <span class="dt">annotation =</span> annotation_airway) <span class="co"># or anno_df_orgdb, or anno_df_biomart</span></a></code></pre></div>
<p>If desired, alternatives can be used. See the well written annotation workflow available at the Bioconductor site (<a href="https://bioconductor.org/help/workflows/annotation/annotation/" class="uri">https://bioconductor.org/help/workflows/annotation/annotation/</a>).</p>
</div>
<div id="running-pcaexplorer-on-synthetic-datasets" class="section level1">
<h1 class="hasAnchor">
<a href="#running-pcaexplorer-on-synthetic-datasets" class="anchor"></a>Running <code>pcaExplorer</code> on synthetic datasets</h1>
<p>For testing and demonstration purposes, a function is also available to generate synthetic datasets whose counts are generated based on two or more experimental factors.</p>
<p>This can be called with the command:</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb21-1" data-line-number="1">dds_multifac &lt;-<span class="st"> </span><span class="kw"><a href="../reference/makeExampleDESeqDataSet_multifac.html">makeExampleDESeqDataSet_multifac</a></span>(<span class="dt">betaSD_condition =</span> <span class="dv">3</span>,<span class="dt">betaSD_tissue =</span> <span class="dv">1</span>)</a></code></pre></div>
<p>See all the available parameters by typing <code><a href="../reference/makeExampleDESeqDataSet_multifac.html">?makeExampleDESeqDataSet_multifac</a></code>. Credits are given to the initial implementation by Mike Love in the <em><a href="https://bioconductor.org/packages/3.9/DESeq2">DESeq2</a></em> package.</p>
<p>The following steps run the app with the synthetic dataset.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb22-1" data-line-number="1">dds_multifac &lt;-<span class="st"> </span><span class="kw"><a href="../reference/makeExampleDESeqDataSet_multifac.html">makeExampleDESeqDataSet_multifac</a></span>(<span class="dt">betaSD_condition =</span> <span class="dv">1</span>,<span class="dt">betaSD_tissue =</span> <span class="dv">3</span>)</a>
<a class="sourceLine" id="cb22-2" data-line-number="2">dds_multifac</a></code></pre></div>
<pre><code>class: DESeqDataSet 
dim: 1000 12 
metadata(1): version
assays(1): counts
rownames(1000): gene1 gene2 ... gene999 gene1000
rowData names(4): trueIntercept trueBeta_condition trueBeta_tissue
  trueDisp
colnames(12): sample1 sample2 ... sample11 sample12
colData names(2): condition tissue</code></pre>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb24-1" data-line-number="1">rld_multifac &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/DESeq2/topics/rlog">rlogTransformation</a></span>(dds_multifac)</a>
<a class="sourceLine" id="cb24-2" data-line-number="2">rld_multifac</a></code></pre></div>
<pre><code>class: DESeqTransform 
dim: 1000 12 
metadata(1): version
assays(1): ''
rownames(1000): gene1 gene2 ... gene999 gene1000
rowData names(11): trueIntercept trueBeta_condition ... dispFit
  rlogIntercept
colnames(12): sample1 sample2 ... sample11 sample12
colData names(3): condition tissue sizeFactor</code></pre>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb26-1" data-line-number="1"><span class="co">## checking how the samples cluster on the PCA plot</span></a>
<a class="sourceLine" id="cb26-2" data-line-number="2"><span class="kw"><a href="../reference/pcaplot.html">pcaplot</a></span>(rld_multifac,<span class="dt">intgroup =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"condition"</span>,<span class="st">"tissue"</span>))</a></code></pre></div>
<p><img src="pcaExplorer_files/figure-html/prepsimul-1.png" width="700" style="display: block; margin: auto;"></p>
<p>Launch the app for exploring this dataset with:</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb27-1" data-line-number="1"><span class="kw"><a href="../reference/pcaExplorer.html">pcaExplorer</a></span>(<span class="dt">dds =</span> dds_multifac,</a>
<a class="sourceLine" id="cb27-2" data-line-number="2">            <span class="dt">dst =</span> rld_multifac)</a></code></pre></div>
<p>When such a dataset is provided, the panel for multifactorial exploration is also usable at its best.</p>
</div>
<div id="functions-exported-by-the-package-for-standalone-usage" class="section level1">
<h1 class="hasAnchor">
<a href="#functions-exported-by-the-package-for-standalone-usage" class="anchor"></a>Functions exported by the package for standalone usage</h1>
<p>The functions exported by the <em><a href="https://bioconductor.org/packages/3.9/pcaExplorer">pcaExplorer</a></em> package can be also used in a standalone scenario, provided the required objects are in the working environment. They are listed here for an overview, but please refer to the documentation for additional details. Where possible, for each function a code snippet will be provided for its typical usage.</p>
<div id="pcaplot" class="section level2">
<h2 class="hasAnchor">
<a href="#pcaplot" class="anchor"></a><code>pcaplot</code>
</h2>
<p><code>pcaplot</code> plots the sample PCA for <code>DESeqTransform</code> objects, such as rlog-transformed data. This is the workhorse of the Samples View tab.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb28-1" data-line-number="1"><span class="kw"><a href="../reference/pcaplot.html">pcaplot</a></span>(rld_airway,<span class="dt">intgroup =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"cell"</span>,<span class="st">"dex"</span>),<span class="dt">ntop =</span> <span class="dv">1000</span>,</a>
<a class="sourceLine" id="cb28-2" data-line-number="2">        <span class="dt">pcX =</span> <span class="dv">1</span>, <span class="dt">pcY =</span> <span class="dv">2</span>, <span class="dt">title =</span> <span class="st">"airway dataset PCA on samples - PC1 vs PC2"</span>)</a></code></pre></div>
<p><img src="pcaExplorer_files/figure-html/func-pcaplot-1.png" width="700" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb29-1" data-line-number="1"><span class="co"># on a different set of principal components...</span></a>
<a class="sourceLine" id="cb29-2" data-line-number="2"><span class="kw"><a href="../reference/pcaplot.html">pcaplot</a></span>(rld_airway,<span class="dt">intgroup =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"dex"</span>),<span class="dt">ntop =</span> <span class="dv">1000</span>,</a>
<a class="sourceLine" id="cb29-3" data-line-number="3">        <span class="dt">pcX =</span> <span class="dv">1</span>, <span class="dt">pcY =</span> <span class="dv">4</span>, <span class="dt">title =</span> <span class="st">"airway dataset PCA on samples - PC1 vs PC4"</span>,</a>
<a class="sourceLine" id="cb29-4" data-line-number="4">        <span class="dt">ellipse =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<p><img src="pcaExplorer_files/figure-html/func-pcaplot-2.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="pcaplot3d" class="section level2">
<h2 class="hasAnchor">
<a href="#pcaplot3d" class="anchor"></a><code>pcaplot3d</code>
</h2>
<p>Same as for <code>pcaplot</code>, but it uses the <code>threejs</code> package for the 3d interactive view.</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb30-1" data-line-number="1"><span class="kw"><a href="../reference/pcaplot3d.html">pcaplot3d</a></span>(rld_airway,<span class="dt">intgroup =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"cell"</span>,<span class="st">"dex"</span>),<span class="dt">ntop =</span> <span class="dv">1000</span>,</a>
<a class="sourceLine" id="cb30-2" data-line-number="2">        <span class="dt">pcX =</span> <span class="dv">1</span>, <span class="dt">pcY =</span> <span class="dv">2</span>, <span class="dt">pcZ =</span> <span class="dv">3</span>)</a>
<a class="sourceLine" id="cb30-3" data-line-number="3"><span class="co"># will open up in the viewer</span></a></code></pre></div>
</div>
<div id="pcascree" class="section level2">
<h2 class="hasAnchor">
<a href="#pcascree" class="anchor"></a><code>pcascree</code>
</h2>
<p><code>pcascree</code> produces a scree plot of the PC computed on the samples. A <code>prcomp</code> object needs to be passed as main argument.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb31-1" data-line-number="1">pcaobj_airway &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/stats/topics/prcomp">prcomp</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/t">t</a></span>(<span class="kw">assay</span>(rld_airway)))</a>
<a class="sourceLine" id="cb31-2" data-line-number="2"><span class="kw"><a href="../reference/pcascree.html">pcascree</a></span>(pcaobj_airway,<span class="dt">type=</span><span class="st">"pev"</span>,</a>
<a class="sourceLine" id="cb31-3" data-line-number="3">         <span class="dt">title=</span><span class="st">"Proportion of explained proportion of variance - airway dataset"</span>)</a></code></pre></div>
<p><img src="pcaExplorer_files/figure-html/func-pcascree-1.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="correlatepcs-and-plotpccorrs" class="section level2">
<h2 class="hasAnchor">
<a href="#correlatepcs-and-plotpccorrs" class="anchor"></a><code>correlatePCs</code> and <code>plotPCcorrs</code>
</h2>
<p><code>correlatePCs</code> and <code>plotPCcorrs</code> respectively compute and plot significance of the (cor)relation of each covariate versus a principal component. The input for <code>correlatePCs</code> is a <code>prcomp</code> object.</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb32-1" data-line-number="1">res_pcairway &lt;-<span class="st"> </span><span class="kw"><a href="../reference/correlatePCs.html">correlatePCs</a></span>(pcaobj_airway,<span class="kw">colData</span>(dds_airway))</a>
<a class="sourceLine" id="cb32-2" data-line-number="2"></a>
<a class="sourceLine" id="cb32-3" data-line-number="3">res_pcairway</a></code></pre></div>
<pre><code>     SampleName       cell        dex albut       Run avgLength Experiment
PC_1  0.4288799 0.68227033 0.02092134    NA 0.4288799 0.2554109  0.4288799
PC_2  0.4288799 0.11161023 0.56370286    NA 0.4288799 0.1993592  0.4288799
PC_3  0.4288799 0.10377716 0.38647623    NA 0.4288799 0.1864725  0.4288799
PC_4  0.4288799 0.08331631 0.56370286    NA 0.4288799 0.4635148  0.4288799
        Sample BioSample
PC_1 0.4288799 0.4288799
PC_2 0.4288799 0.4288799
PC_3 0.4288799 0.4288799
PC_4 0.4288799 0.4288799</code></pre>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb34-1" data-line-number="1"><span class="kw"><a href="../reference/plotPCcorrs.html">plotPCcorrs</a></span>(res_pcairway)</a></code></pre></div>
<p><img src="pcaExplorer_files/figure-html/func-correlatepcs-1.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="hi_loadings" class="section level2">
<h2 class="hasAnchor">
<a href="#hi_loadings" class="anchor"></a><code>hi_loadings</code>
</h2>
<p><code>hi_loadings</code> extracts and optionally plots the genes with the highest loadings.</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb35-1" data-line-number="1"><span class="co"># extract the table of the genes with high loadings</span></a>
<a class="sourceLine" id="cb35-2" data-line-number="2"><span class="kw"><a href="../reference/hi_loadings.html">hi_loadings</a></span>(pcaobj_airway,<span class="dt">topN =</span> <span class="dv">10</span>,<span class="dt">exprTable=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/DESeq2/topics/counts">counts</a></span>(dds_airway))</a></code></pre></div>
<pre><code>                SRR1039508 SRR1039509 SRR1039512 SRR1039513 SRR1039516
ENSG00000143127         11        108         24        485         41
ENSG00000168309         12        274         35        451          1
ENSG00000101347       1632      17126       2098      19694       1598
ENSG00000211445        916      15749       3142      24057       1627
ENSG00000096060        260       4652        381       3875        601
ENSG00000163884         70       1325         52        702         36
ENSG00000171819          4         50         19        543          1
ENSG00000127954         13        247         25        889          2
ENSG00000152583         62       2040         99       1172        100
ENSG00000109906          4        739          5        429          1
ENSG00000162692        914         62       1192         55       1359
ENSG00000178695       4746        830       4805        414       5321
ENSG00000214814        312         24        193         28        501
ENSG00000164742       1506        347        275         14        137
ENSG00000138316       1327        207       1521        118       1962
ENSG00000123610        444        136        303         36       1170
ENSG00000124766       2483        406       2057        185       2829
ENSG00000105989        562         47       1575        106        106
ENSG00000013293        268         23        435         56        558
ENSG00000146250        330         41        907         89        720
                SRR1039517 SRR1039520 SRR1039521
ENSG00000143127        607         77        660
ENSG00000168309         65          4        193
ENSG00000101347      17697       1683      32036
ENSG00000211445      16274       1741      24883
ENSG00000096060       5493        154       4118
ENSG00000163884        487         34       1355
ENSG00000171819         10         14       1067
ENSG00000127954        199         20        462
ENSG00000152583       1924         79       2138
ENSG00000109906        581         12       1113
ENSG00000162692        171        646         31
ENSG00000178695       1391       4411        606
ENSG00000214814         65        789         76
ENSG00000164742         37        475         56
ENSG00000138316        618       1045        152
ENSG00000123610        195        473         37
ENSG00000124766        870       1851        301
ENSG00000105989         24        382         46
ENSG00000013293         75        562         74
ENSG00000146250        123        439         60</code></pre>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb37-1" data-line-number="1"><span class="co"># or alternatively plot the values</span></a>
<a class="sourceLine" id="cb37-2" data-line-number="2"><span class="kw"><a href="../reference/hi_loadings.html">hi_loadings</a></span>(pcaobj_airway,<span class="dt">topN =</span> <span class="dv">10</span>,<span class="dt">annotation =</span> annotation_airway)</a></code></pre></div>
<p><img src="pcaExplorer_files/figure-html/func-hiloadings-1.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="genespca" class="section level2">
<h2 class="hasAnchor">
<a href="#genespca" class="anchor"></a><code>genespca</code>
</h2>
<p><code>genespca</code> computes and plots the principal components of the genes, eventually displaying the samples as in a typical biplot visualization. This is the function in action for the Genes View tab.</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb38-1" data-line-number="1">groups_airway &lt;-<span class="st"> </span><span class="kw">colData</span>(dds_airway)<span class="op">$</span>dex</a>
<a class="sourceLine" id="cb38-2" data-line-number="2">cols_airway &lt;-<span class="st"> </span>scales<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/scales/topics/hue_pal">hue_pal</a></span>()(<span class="dv">2</span>)[groups_airway]</a>
<a class="sourceLine" id="cb38-3" data-line-number="3"><span class="co"># with many genes, do not plot the labels of the genes</span></a>
<a class="sourceLine" id="cb38-4" data-line-number="4"><span class="kw"><a href="../reference/genespca.html">genespca</a></span>(rld_airway,<span class="dt">ntop=</span><span class="dv">5000</span>,</a>
<a class="sourceLine" id="cb38-5" data-line-number="5">         <span class="dt">choices =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">1</span>,<span class="dv">2</span>),</a>
<a class="sourceLine" id="cb38-6" data-line-number="6">         <span class="dt">arrowColors=</span>cols_airway,<span class="dt">groupNames=</span>groups_airway,</a>
<a class="sourceLine" id="cb38-7" data-line-number="7">         <span class="dt">alpha =</span> <span class="fl">0.2</span>,</a>
<a class="sourceLine" id="cb38-8" data-line-number="8">         <span class="dt">useRownamesAsLabels=</span><span class="ot">FALSE</span>,</a>
<a class="sourceLine" id="cb38-9" data-line-number="9">         <span class="dt">varname.size =</span> <span class="dv">5</span></a>
<a class="sourceLine" id="cb38-10" data-line-number="10">        )</a></code></pre></div>
<p><img src="pcaExplorer_files/figure-html/func-genespca-1.png" width="700" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb39-1" data-line-number="1"><span class="co"># with a smaller number of genes, plot gene names included in the annotation</span></a>
<a class="sourceLine" id="cb39-2" data-line-number="2"><span class="kw"><a href="../reference/genespca.html">genespca</a></span>(rld_airway,<span class="dt">ntop=</span><span class="dv">100</span>,</a>
<a class="sourceLine" id="cb39-3" data-line-number="3">         <span class="dt">choices =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">1</span>,<span class="dv">2</span>),</a>
<a class="sourceLine" id="cb39-4" data-line-number="4">         <span class="dt">arrowColors=</span>cols_airway,<span class="dt">groupNames=</span>groups_airway,</a>
<a class="sourceLine" id="cb39-5" data-line-number="5">         <span class="dt">alpha =</span> <span class="fl">0.7</span>,</a>
<a class="sourceLine" id="cb39-6" data-line-number="6">         <span class="dt">varname.size =</span> <span class="dv">5</span>,</a>
<a class="sourceLine" id="cb39-7" data-line-number="7">         <span class="dt">annotation =</span> annotation_airway</a>
<a class="sourceLine" id="cb39-8" data-line-number="8">        )</a></code></pre></div>
<p><img src="pcaExplorer_files/figure-html/func-genespca-2.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="topgotable" class="section level2">
<h2 class="hasAnchor">
<a href="#topgotable" class="anchor"></a><code>topGOtable</code>
</h2>
<p><code>topGOtable</code> is a convenient wrapper for extracting functional GO terms enriched in a subset of genes (such as the differentially expressed genes), based on the algorithm and the implementation in the <em><a href="https://bioconductor.org/packages/3.9/topGO">topGO</a></em> package.</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb40-1" data-line-number="1"><span class="co"># example not run due to quite long runtime</span></a>
<a class="sourceLine" id="cb40-2" data-line-number="2">dds_airway &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/DESeq2/topics/DESeq">DESeq</a></span>(dds_airway)</a>
<a class="sourceLine" id="cb40-3" data-line-number="3">res_airway &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/DESeq2/topics/results">results</a></span>(dds_airway)</a>
<a class="sourceLine" id="cb40-4" data-line-number="4">res_airway<span class="op">$</span>symbol &lt;-<span class="st"> </span><span class="kw">mapIds</span>(org.Hs.eg.db,</a>
<a class="sourceLine" id="cb40-5" data-line-number="5">                            <span class="dt">keys=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/row.names">row.names</a></span>(res_airway),</a>
<a class="sourceLine" id="cb40-6" data-line-number="6">                            <span class="dt">column=</span><span class="st">"SYMBOL"</span>,</a>
<a class="sourceLine" id="cb40-7" data-line-number="7">                            <span class="dt">keytype=</span><span class="st">"ENSEMBL"</span>,</a>
<a class="sourceLine" id="cb40-8" data-line-number="8">                            <span class="dt">multiVals=</span><span class="st">"first"</span>)</a>
<a class="sourceLine" id="cb40-9" data-line-number="9">res_airway<span class="op">$</span>entrez &lt;-<span class="st"> </span><span class="kw">mapIds</span>(org.Hs.eg.db,</a>
<a class="sourceLine" id="cb40-10" data-line-number="10">                            <span class="dt">keys=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/row.names">row.names</a></span>(res_airway),</a>
<a class="sourceLine" id="cb40-11" data-line-number="11">                            <span class="dt">column=</span><span class="st">"ENTREZID"</span>,</a>
<a class="sourceLine" id="cb40-12" data-line-number="12">                            <span class="dt">keytype=</span><span class="st">"ENSEMBL"</span>,</a>
<a class="sourceLine" id="cb40-13" data-line-number="13">                            <span class="dt">multiVals=</span><span class="st">"first"</span>)</a>
<a class="sourceLine" id="cb40-14" data-line-number="14">resOrdered &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/as.data.frame">as.data.frame</a></span>(res_airway[<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/order">order</a></span>(res_airway<span class="op">$</span>padj),])</a>
<a class="sourceLine" id="cb40-15" data-line-number="15"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/head">head</a></span>(resOrdered)</a>
<a class="sourceLine" id="cb40-16" data-line-number="16"><span class="co"># extract DE genes</span></a>
<a class="sourceLine" id="cb40-17" data-line-number="17">de_df &lt;-<span class="st"> </span>resOrdered[resOrdered<span class="op">$</span>padj <span class="op">&lt;</span><span class="st"> </span><span class="fl">.05</span> <span class="op">&amp;</span><span class="st"> </span><span class="op">!</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/NA">is.na</a></span>(resOrdered<span class="op">$</span>padj),]</a>
<a class="sourceLine" id="cb40-18" data-line-number="18">de_symbols &lt;-<span class="st"> </span>de_df<span class="op">$</span>symbol</a>
<a class="sourceLine" id="cb40-19" data-line-number="19"><span class="co"># extract background genes</span></a>
<a class="sourceLine" id="cb40-20" data-line-number="20">bg_ids &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">rownames</a></span>(dds_airway)[<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colSums">rowSums</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/DESeq2/topics/counts">counts</a></span>(dds_airway)) <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>]</a>
<a class="sourceLine" id="cb40-21" data-line-number="21">bg_symbols &lt;-<span class="st"> </span><span class="kw">mapIds</span>(org.Hs.eg.db,</a>
<a class="sourceLine" id="cb40-22" data-line-number="22">                     <span class="dt">keys=</span>bg_ids,</a>
<a class="sourceLine" id="cb40-23" data-line-number="23">                     <span class="dt">column=</span><span class="st">"SYMBOL"</span>,</a>
<a class="sourceLine" id="cb40-24" data-line-number="24">                     <span class="dt">keytype=</span><span class="st">"ENSEMBL"</span>,</a>
<a class="sourceLine" id="cb40-25" data-line-number="25">                     <span class="dt">multiVals=</span><span class="st">"first"</span>)</a>
<a class="sourceLine" id="cb40-26" data-line-number="26"><span class="co"># run the function</span></a>
<a class="sourceLine" id="cb40-27" data-line-number="27">topgoDE_airway &lt;-<span class="st"> </span><span class="kw"><a href="../reference/topGOtable.html">topGOtable</a></span>(de_symbols, bg_symbols,</a>
<a class="sourceLine" id="cb40-28" data-line-number="28">                             <span class="dt">ontology =</span> <span class="st">"BP"</span>,</a>
<a class="sourceLine" id="cb40-29" data-line-number="29">                             <span class="dt">mapping =</span> <span class="st">"org.Hs.eg.db"</span>,</a>
<a class="sourceLine" id="cb40-30" data-line-number="30">                             <span class="dt">geneID =</span> <span class="st">"symbol"</span>)</a></code></pre></div>
</div>
<div id="pca2go-docs" class="section level2">
<h2 class="hasAnchor">
<a href="#pca2go-docs" class="anchor"></a><code>pca2go</code>
</h2>
<p><code>pca2go</code> provides a functional interpretation of the principal components, by extracting the genes with the highest loadings for each PC, and then runs internally <code>topGOtable</code> on them for efficient functional enrichment analysis. This function requires a <code>DESeqTransform</code> object as main parameter.</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb41-1" data-line-number="1">pca2go_airway &lt;-<span class="st"> </span><span class="kw"><a href="../reference/pca2go.html">pca2go</a></span>(rld_airway,</a>
<a class="sourceLine" id="cb41-2" data-line-number="2">                        <span class="dt">annotation =</span> annotation_airway,</a>
<a class="sourceLine" id="cb41-3" data-line-number="3">                        <span class="dt">organism =</span> <span class="st">"Hs"</span>,</a>
<a class="sourceLine" id="cb41-4" data-line-number="4">                        <span class="dt">ensToGeneSymbol =</span> <span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb41-5" data-line-number="5">                        <span class="dt">background_genes =</span> bg_ids)</a>
<a class="sourceLine" id="cb41-6" data-line-number="6"><span class="co"># for a smooth interactive exploration, use DT::datatable</span></a>
<a class="sourceLine" id="cb41-7" data-line-number="7"><span class="kw">datatable</span>(pca2go_airway<span class="op">$</span>PC1<span class="op">$</span>posLoad)</a>
<a class="sourceLine" id="cb41-8" data-line-number="8"><span class="co"># display it in the normal R session...</span></a>
<a class="sourceLine" id="cb41-9" data-line-number="9"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/head">head</a></span>(pca2go_airway<span class="op">$</span>PC1<span class="op">$</span>posLoad)</a>
<a class="sourceLine" id="cb41-10" data-line-number="10"><span class="co"># ... or use it for running the app and display in the dedicated tab</span></a>
<a class="sourceLine" id="cb41-11" data-line-number="11"><span class="kw"><a href="../reference/pcaExplorer.html">pcaExplorer</a></span>(dds_airway,rld_airway,</a>
<a class="sourceLine" id="cb41-12" data-line-number="12">            <span class="dt">pca2go =</span> pca2go_airway,</a>
<a class="sourceLine" id="cb41-13" data-line-number="13">            <span class="dt">annotation =</span> annotation_airway)</a></code></pre></div>
</div>
<div id="limmaquickpca2go" class="section level2">
<h2 class="hasAnchor">
<a href="#limmaquickpca2go" class="anchor"></a><code>limmaquickpca2go</code>
</h2>
<p><code>limmaquickpca2go</code> is an alternative to <code>pca2go</code>, used in the live running app, thanks to its fast implementation based on the <code><a href="https://www.rdocumentation.org/packages/limma/topics/goana">limma::goana</a></code> function.</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb42-1" data-line-number="1">goquick_airway &lt;-<span class="st"> </span><span class="kw"><a href="../reference/limmaquickpca2go.html">limmaquickpca2go</a></span>(rld_airway,</a>
<a class="sourceLine" id="cb42-2" data-line-number="2">                                   <span class="dt">pca_ngenes =</span> <span class="dv">10000</span>,</a>
<a class="sourceLine" id="cb42-3" data-line-number="3">                                   <span class="dt">inputType =</span> <span class="st">"ENSEMBL"</span>,</a>
<a class="sourceLine" id="cb42-4" data-line-number="4">                                   <span class="dt">organism =</span> <span class="st">"Hs"</span>)</a>
<a class="sourceLine" id="cb42-5" data-line-number="5"><span class="co"># display it in the normal R session...</span></a>
<a class="sourceLine" id="cb42-6" data-line-number="6"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/head">head</a></span>(goquick_airway<span class="op">$</span>PC1<span class="op">$</span>posLoad)</a>
<a class="sourceLine" id="cb42-7" data-line-number="7"><span class="co"># ... or use it for running the app and display in the dedicated tab</span></a>
<a class="sourceLine" id="cb42-8" data-line-number="8"><span class="kw"><a href="../reference/pcaExplorer.html">pcaExplorer</a></span>(dds_airway,rld_airway,</a>
<a class="sourceLine" id="cb42-9" data-line-number="9">            <span class="dt">pca2go =</span> goquick_airway,</a>
<a class="sourceLine" id="cb42-10" data-line-number="10">            <span class="dt">annotation =</span> annotation_airway)</a></code></pre></div>
</div>
<div id="makeexampledeseqdataset_multifac" class="section level2">
<h2 class="hasAnchor">
<a href="#makeexampledeseqdataset_multifac" class="anchor"></a><code>makeExampleDESeqDataSet_multifac</code>
</h2>
<p><code>makeExampleDESeqDataSet_multifac</code> constructs a simulated <code>DESeqDataSet</code> of Negative Binomial dataset from different conditions. The fold changes between the conditions can be adjusted with the <code>betaSD_condition</code> and <code>betaSD_tissue</code> arguments.</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb43-1" data-line-number="1">dds_simu &lt;-<span class="st"> </span><span class="kw"><a href="../reference/makeExampleDESeqDataSet_multifac.html">makeExampleDESeqDataSet_multifac</a></span>(<span class="dt">betaSD_condition =</span> <span class="dv">3</span>,<span class="dt">betaSD_tissue =</span> <span class="fl">0.5</span>)</a>
<a class="sourceLine" id="cb43-2" data-line-number="2">dds_simu</a></code></pre></div>
<pre><code>class: DESeqDataSet 
dim: 1000 12 
metadata(1): version
assays(1): counts
rownames(1000): gene1 gene2 ... gene999 gene1000
rowData names(4): trueIntercept trueBeta_condition trueBeta_tissue
  trueDisp
colnames(12): sample1 sample2 ... sample11 sample12
colData names(2): condition tissue</code></pre>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb45-1" data-line-number="1">dds2_simu &lt;-<span class="st"> </span><span class="kw"><a href="../reference/makeExampleDESeqDataSet_multifac.html">makeExampleDESeqDataSet_multifac</a></span>(<span class="dt">betaSD_condition =</span> <span class="fl">0.5</span>,<span class="dt">betaSD_tissue =</span> <span class="dv">2</span>)</a>
<a class="sourceLine" id="cb45-2" data-line-number="2">dds2_simu</a></code></pre></div>
<pre><code>class: DESeqDataSet 
dim: 1000 12 
metadata(1): version
assays(1): counts
rownames(1000): gene1 gene2 ... gene999 gene1000
rowData names(4): trueIntercept trueBeta_condition trueBeta_tissue
  trueDisp
colnames(12): sample1 sample2 ... sample11 sample12
colData names(2): condition tissue</code></pre>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb47-1" data-line-number="1">rld_simu &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/DESeq2/topics/rlog">rlogTransformation</a></span>(dds_simu)</a>
<a class="sourceLine" id="cb47-2" data-line-number="2">rld2_simu &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/DESeq2/topics/rlog">rlogTransformation</a></span>(dds2_simu)</a>
<a class="sourceLine" id="cb47-3" data-line-number="3"><span class="kw"><a href="../reference/pcaplot.html">pcaplot</a></span>(rld_simu,<span class="dt">intgroup =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"condition"</span>,<span class="st">"tissue"</span>)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb47-4" data-line-number="4"><span class="st">  </span>ggplot2<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/ggplot2/topics/labs">ggtitle</a></span>(<span class="st">"Simulated data - Big condition effect, small tissue effect"</span>)</a></code></pre></div>
<p><img src="pcaExplorer_files/figure-html/func-makedataset-1.png" width="700" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb48-1" data-line-number="1"><span class="kw"><a href="../reference/pcaplot.html">pcaplot</a></span>(rld2_simu,<span class="dt">intgroup =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"condition"</span>,<span class="st">"tissue"</span>)) <span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb48-2" data-line-number="2"><span class="st">  </span>ggplot2<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/ggplot2/topics/labs">ggtitle</a></span>(<span class="st">"Simulated data - Small condition effect, bigger tissue effect"</span>)</a></code></pre></div>
<p><img src="pcaExplorer_files/figure-html/func-makedataset-2.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="distro_expr" class="section level2">
<h2 class="hasAnchor">
<a href="#distro_expr" class="anchor"></a><code>distro_expr</code>
</h2>
<p>Plots the distribution of expression values, either with density lines, boxplots or violin plots.</p>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb49-1" data-line-number="1"><span class="kw"><a href="../reference/distro_expr.html">distro_expr</a></span>(rld_airway,<span class="dt">plot_type =</span> <span class="st">"density"</span>)</a>
<a class="sourceLine" id="cb49-2" data-line-number="2"><span class="kw"><a href="../reference/distro_expr.html">distro_expr</a></span>(rld_airway,<span class="dt">plot_type =</span> <span class="st">"violin"</span>)</a></code></pre></div>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb50-1" data-line-number="1"><span class="kw"><a href="../reference/distro_expr.html">distro_expr</a></span>(rld_airway,<span class="dt">plot_type =</span> <span class="st">"boxplot"</span>)</a></code></pre></div>
<p><img src="pcaExplorer_files/figure-html/unnamed-chunk-6-1.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="geneprofiler" class="section level2">
<h2 class="hasAnchor">
<a href="#geneprofiler" class="anchor"></a><code>geneprofiler</code>
</h2>
<p>Plots the profile expression of a subset of genes, optionally as standardized values.</p>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb51-1" data-line-number="1">dds &lt;-<span class="st"> </span><span class="kw"><a href="../reference/makeExampleDESeqDataSet_multifac.html">makeExampleDESeqDataSet_multifac</a></span>(<span class="dt">betaSD_condition =</span> <span class="dv">3</span>,<span class="dt">betaSD_tissue =</span> <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb51-2" data-line-number="2">dst &lt;-<span class="st"> </span>DESeq2<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/DESeq2/topics/rlog">rlogTransformation</a></span>(dds)</a>
<a class="sourceLine" id="cb51-3" data-line-number="3"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/Random">set.seed</a></span>(<span class="dv">42</span>)</a>
<a class="sourceLine" id="cb51-4" data-line-number="4"><span class="kw"><a href="../reference/geneprofiler.html">geneprofiler</a></span>(dst,<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/paste">paste0</a></span>(<span class="st">"gene"</span>,<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/sample">sample</a></span>(<span class="dv">1</span><span class="op">:</span><span class="dv">1000</span>,<span class="dv">20</span>)), <span class="dt">plotZ =</span> <span class="ot">FALSE</span>)</a></code></pre></div>
<pre><code>you provided 20 unique identifiers
20 out of 20 provided genes were found in the data</code></pre>
<p><img src="pcaExplorer_files/figure-html/unnamed-chunk-7-1.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div id="get_annotation-and-get_annotation_orgdb" class="section level2">
<h2 class="hasAnchor">
<a href="#get_annotation-and-get_annotation_orgdb" class="anchor"></a><code>get_annotation</code> and <code>get_annotation_orgdb</code>
</h2>
<p>These two wrapper functions retrieve the latest annotations for the <code>dds</code> object, to be used in the call to the <code>pcaExplorer</code> function. They use respectively the <code>biomaRt</code> package and the <code>org.XX.eg.db</code> packages.</p>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb53-1" data-line-number="1">anno_df_biomart &lt;-<span class="st"> </span><span class="kw"><a href="../reference/get_annotation.html">get_annotation</a></span>(<span class="dt">dds =</span> dds_airway,</a>
<a class="sourceLine" id="cb53-2" data-line-number="2">                                  <span class="dt">biomart_dataset =</span> <span class="st">"hsapiens_gene_ensembl"</span>,</a>
<a class="sourceLine" id="cb53-3" data-line-number="3">                                  <span class="dt">idtype =</span> <span class="st">"ensembl_gene_id"</span>)</a>
<a class="sourceLine" id="cb53-4" data-line-number="4"></a>
<a class="sourceLine" id="cb53-5" data-line-number="5">anno_df_orgdb &lt;-<span class="st"> </span><span class="kw"><a href="../reference/get_annotation_orgdb.html">get_annotation_orgdb</a></span>(<span class="dt">dds =</span> dds_airway,</a>
<a class="sourceLine" id="cb53-6" data-line-number="6">                                      <span class="dt">orgdb_species =</span> <span class="st">"org.Hs.eg.db"</span>,</a>
<a class="sourceLine" id="cb53-7" data-line-number="7">                                      <span class="dt">idtype =</span> <span class="st">"ENSEMBL"</span>)</a></code></pre></div>
</div>
<div id="pair_corr" class="section level2">
<h2 class="hasAnchor">
<a href="#pair_corr" class="anchor"></a><code>pair_corr</code>
</h2>
<p>Plots the pairwise scatter plots and computes the correlation coefficient on the expression matrix provided.</p>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb54-1" data-line-number="1"><span class="co"># use a subset of the counts to reduce plotting time, it can be time consuming with many samples</span></a>
<a class="sourceLine" id="cb54-2" data-line-number="2"><span class="kw"><a href="../reference/pair_corr.html">pair_corr</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/DESeq2/topics/counts">counts</a></span>(dds_airway)[<span class="dv">1</span><span class="op">:</span><span class="dv">100</span>,])</a></code></pre></div>
<p><img src="pcaExplorer_files/figure-html/unnamed-chunk-9-1.png" width="700" style="display: block; margin: auto;"></p>
</div>
</div>
<div id="further-development" class="section level1">
<h1 class="hasAnchor">
<a href="#further-development" class="anchor"></a>Further development</h1>
<p>Additional functionality for the <em><a href="https://bioconductor.org/packages/3.9/pcaExplorer">pcaExplorer</a></em> will be added in the future, as it is tightly related to a topic under current development research.</p>
<p>Improvements, suggestions, bugs, issues and feedback of any type can be sent to <a href="mailto:marinif@uni-mainz.de" class="email">marinif@uni-mainz.de</a>.</p>
</div>
<div id="session-info" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h1>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb55-1" data-line-number="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/sessionInfo">sessionInfo</a></span>()</a></code></pre></div>
<pre><code>R Under development (unstable) (2018-11-21 r75637)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Sierra 10.12.6

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] org.Hs.eg.db_3.7.0          AnnotationDbi_1.45.1       
 [3] DESeq2_1.23.8               airway_1.3.0               
 [5] SummarizedExperiment_1.13.0 DelayedArray_0.9.8         
 [7] BiocParallel_1.17.17        matrixStats_0.54.0         
 [9] GenomicRanges_1.35.1        GenomeInfoDb_1.19.2        
[11] IRanges_2.17.4              S4Vectors_0.21.16          
[13] pcaExplorer_2.9.5           bigmemory_4.5.33           
[15] Biobase_2.43.1              BiocGenerics_0.29.2        
[17] knitr_1.22                  BiocStyle_2.11.0           

loaded via a namespace (and not attached):
  [1] GOstats_2.49.0         backports_1.1.3        Hmisc_4.2-0           
  [4] NMF_0.21.0             plyr_1.8.4             igraph_1.2.4          
  [7] lazyeval_0.2.2         GSEABase_1.45.0        shinydashboard_0.7.1  
 [10] splines_3.6.0          crosstalk_1.0.0        ggplot2_3.1.0         
 [13] gridBase_0.4-7         digest_0.6.18          foreach_1.4.4         
 [16] htmltools_0.3.6        GO.db_3.7.0            magrittr_1.5          
 [19] checkmate_1.9.1        memoise_1.1.0          cluster_2.0.7-1       
 [22] doParallel_1.0.14      limma_3.39.12          annotate_1.61.1       
 [25] pkgdown_1.3.0          prettyunits_1.0.2      colorspace_1.4-1      
 [28] blob_1.1.1             ggrepel_0.8.0          xfun_0.5              
 [31] dplyr_0.8.0.1          crayon_1.3.4           RCurl_1.95-4.12       
 [34] bigmemory.sri_0.1.3    graph_1.61.0           roxygen2_6.1.1        
 [37] genefilter_1.65.0      survival_2.43-3        iterators_1.0.10      
 [40] glue_1.3.1             registry_0.5-1         gtable_0.2.0          
 [43] zlibbioc_1.29.0        XVector_0.23.2         Rgraphviz_2.27.0      
 [46] SparseM_1.77           scales_1.0.0           pheatmap_1.0.12       
 [49] DBI_1.0.0              rngtools_1.3.1         bibtex_0.4.2          
 [52] Rcpp_1.0.1             xtable_1.8-3           progress_1.2.0        
 [55] htmlTable_1.13.1       foreign_0.8-71         bit_1.1-14            
 [58] Formula_1.2-3          DT_0.5                 AnnotationForge_1.25.0
 [61] htmlwidgets_1.3        httr_1.4.0             threejs_0.3.1         
 [64] RColorBrewer_1.1-2     shinyAce_0.3.3         acepack_1.4.1         
 [67] pkgconfig_2.0.2        XML_3.98-1.19          nnet_7.3-12           
 [70] locfit_1.5-9.1         labeling_0.3           tidyselect_0.2.5      
 [73] rlang_0.3.2            reshape2_1.4.3         later_0.8.0           
 [76] munsell_0.5.0          tools_3.6.0            RSQLite_2.1.1         
 [79] evaluate_0.13          shinyBS_0.61           stringr_1.4.0         
 [82] yaml_2.2.0             bit64_0.9-7            fs_1.2.7              
 [85] purrr_0.3.2            RBGL_1.59.1            mime_0.6              
 [88] xml2_1.2.0             biomaRt_2.39.2         compiler_3.6.0        
 [91] rstudioapi_0.10        png_0.1-7              tibble_2.1.1          
 [94] geneplotter_1.61.0     stringi_1.4.3          desc_1.2.0            
 [97] lattice_0.20-38        Matrix_1.2-17          commonmark_1.7        
[100] pillar_1.3.1           BiocManager_1.30.4     d3heatmap_0.6.1.2     
[103] data.table_1.12.0      bitops_1.0-6           httpuv_1.5.0          
[106] R6_2.4.0               latticeExtra_0.6-28    bookdown_0.9          
[109] promises_1.0.1         topGO_2.35.0           gridExtra_2.3         
[112] codetools_0.2-16       MASS_7.3-51.2          assertthat_0.2.1      
[115] Category_2.49.1        pkgmaker_0.27          rprojroot_1.3-2       
[118] withr_2.1.2            GenomeInfoDbData_1.2.0 hms_0.4.2             
[121] grid_3.6.0             rpart_4.1-13           tidyr_0.8.3           
[124] rmarkdown_1.12         shiny_1.2.0            base64enc_0.1-3       </code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#getting-started">Getting started</a></li>
      <li>
<a href="#introduction">Introduction</a><ul class="nav nav-pills nav-stacked">
<li><a href="#citation-info">Citation info</a></li>
      </ul>
</li>
      <li>
<a href="#launching-the-application">Launching the application</a><ul class="nav nav-pills nav-stacked">
<li><a href="#how-to-provide-your-input-data-in-pcaexplorer">How to provide your input data in <em><span>pcaExplorer</span></em></a></li>
      <li><a href="#up-and-running-with-pcaexplorer">Up and running with <em><span>pcaExplorer</span></em></a></li>
      </ul>
</li>
      <li>
<a href="#the-controls-sidebar">The controls sidebar</a><ul class="nav nav-pills nav-stacked">
<li><a href="#app-settings">App settings</a></li>
      <li><a href="#plot-export-settings">Plot export settings</a></li>
      </ul>
</li>
      <li><a href="#the-task-menu">The task menu</a></li>
      <li>
<a href="#the-app-panels">The app panels</a><ul class="nav nav-pills nav-stacked">
<li><a href="#data-upload">Data Upload</a></li>
      <li><a href="#instructions">Instructions</a></li>
      <li><a href="#counts-table">Counts Table</a></li>
      <li><a href="#data-overview">Data Overview</a></li>
      <li><a href="#samples-view">Samples View</a></li>
      <li><a href="#genes-view">Genes View</a></li>
      <li><a href="#genefinder">GeneFinder</a></li>
      <li><a href="#pca2go">PCA2GO</a></li>
      <li><a href="#multifactor-exploration">Multifactor Exploration</a></li>
      <li><a href="#report-editor">Report Editor</a></li>
      <li><a href="#about">About</a></li>
      </ul>
</li>
      <li><a href="#running-pcaexplorer-on-published-datasets">Running <code>pcaExplorer</code> on published datasets</a></li>
      <li><a href="#running-pcaexplorer-on-synthetic-datasets">Running <code>pcaExplorer</code> on synthetic datasets</a></li>
      <li>
<a href="#functions-exported-by-the-package-for-standalone-usage">Functions exported by the package for standalone usage</a><ul class="nav nav-pills nav-stacked">
<li><a href="#pcaplot"><code>pcaplot</code></a></li>
      <li><a href="#pcaplot3d"><code>pcaplot3d</code></a></li>
      <li><a href="#pcascree"><code>pcascree</code></a></li>
      <li><a href="#correlatepcs-and-plotpccorrs"><code>correlatePCs</code> and <code>plotPCcorrs</code></a></li>
      <li><a href="#hi_loadings"><code>hi_loadings</code></a></li>
      <li><a href="#genespca"><code>genespca</code></a></li>
      <li><a href="#topgotable"><code>topGOtable</code></a></li>
      <li><a href="#pca2go-docs"><code>pca2go</code></a></li>
      <li><a href="#limmaquickpca2go"><code>limmaquickpca2go</code></a></li>
      <li><a href="#makeexampledeseqdataset_multifac"><code>makeExampleDESeqDataSet_multifac</code></a></li>
      <li><a href="#distro_expr"><code>distro_expr</code></a></li>
      <li><a href="#geneprofiler"><code>geneprofiler</code></a></li>
      <li><a href="#get_annotation-and-get_annotation_orgdb"><code>get_annotation</code> and <code>get_annotation_orgdb</code></a></li>
      <li><a href="#pair_corr"><code>pair_corr</code></a></li>
      </ul>
</li>
      <li><a href="#further-development">Further development</a></li>
      <li><a href="#session-info">Session info</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by <a href="https://federicomarini.github.io">Federico Marini</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
